enterprise/reporting/event_stats_test.go

```diff
@@ -0,0 +1,128 @@
+	mockBackendConfig.EXPECT().Subscribe(gomock.Any(), backendconfig.TopicBackendConfig).DoAndReturn(func(ctx context.Context, topic backendconfig.Topic) pubsub.DataChannel {
+		ch := make(chan pubsub.DataEvent, 1)
+		ch <- pubsub.DataEvent{
+			Data: map[string]backendconfig.ConfigT{
+				workspaceID: {
+					WorkspaceID: workspaceID,
+					Sources: []backendconfig.SourceT{
+						{
+							ID:      sourceID,
+							Enabled: true,
+							SourceDefinition: backendconfig.SourceDefinitionT{
+								Category: sourceCategory,
+							},
+							Destinations: []backendconfig.DestinationT{
+								{
+									ID:      destinationID,
+									Enabled: true,
+									DestinationDefinition: backendconfig.DestinationDefinitionT{
+										Name: "test-destination-name",
+									},
+								}, // Added a comma here
+							},
+						},
+					},
+				},
+			},
+			Topic: string(backendconfig.TopicBackendConfig),
+		}
+		close(ch)
+		return ch
+	}).AnyTimes()
+
+	statsStore := memstats.New()
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	cs := newConfigSubscriber(logger.NOP)
+
+	subscribeDone := make(chan struct{})
+	go func() {
+		defer close(subscribeDone)
+
+		cs.Subscribe(ctx, mockBackendConfig)
+	}()
+
+	testReports := []*types.PUReportedMetric{
+		{
+			ConnectionDetails: types.ConnectionDetails{
+				SourceID:       sourceID,
+				DestinationID:  destinationID,
+				SourceCategory: sourceCategory,
+			},
+			PUDetails: types.PUDetails{
+				PU:         reportedBy,
+				TerminalPU: true,
+			},
+			StatusDetail: &types.StatusDetail{
+				Count:      10,
+				Status:     "succeeded",
+				StatusCode: 200,
+			},
+		},
+		{
+			ConnectionDetails: types.ConnectionDetails{
+				SourceID:       sourceID,
+				DestinationID:  destinationID,
+				SourceCategory: sourceCategory,
+			},
+			PUDetails: types.PUDetails{
+				PU:         reportedBy,
+				TerminalPU: false,
+			},
+			StatusDetail: &types.StatusDetail{
+				Count:      10,
+				Status:     "aborted",
+				StatusCode: 500,
+			},
+		},
+	}
+	esr := NewEventStatsReporter(cs, statsStore)
+	err := esr.Report(testReports)
+	require.NoError(t, err)
+	require.Equal(t, statsStore.Get(EventsDeliveredMetricName, map[string]string{
+		"workspaceId":     workspaceID,
+		"sourceId":        sourceID,
+		"destinationId":   destinationID,
+		"reportedBy":      reportedBy,
+		"sourceCategory":  sourceCategory,
+		"terminal":        "true",
+		"status_code":     "200",
+		"destinationType": "test-destination-name",
+	}).LastValue(), float64(10))
+	require.Equal(t, statsStore.Get(EventsAbortedMetricName, map[string]string{
+		"workspaceId":     workspaceID,
+		"sourceId":        sourceID,
+		"destinationId":   destinationID,
+		"reportedBy":      reportedBy,
+		"sourceCategory":  sourceCategory,
+		"terminal":        "false",
+		"status_code":     "500",
+		"destinationType": "test-destination-name",
+	}).LastValue(), float64(10))
+}
```

Comment:
We need to wait here for the `subscribeDone` to be closed. 
1. Either we can wait here
```
    <-subscribeDone
```

Or

2. Or it in the t.Cleanup when we are creating?
```
	t.Cleanup(func() {
		cancel()
		<-subscribeDone
	})
```